cdk/stack.ts

```diff
@@ -575,16 +580,26 @@ export class PinBoardStack extends Stack {
     );
     pinboardAppsyncUserTable.grantReadWriteData(usersRefresherLambdaFunction);
 
+    S3.Bucket.fromBucketName(
+      thisStack,
+      PERMISSIONS_BUCKET,
+      PERMISSIONS_BUCKET
+    ).addObjectCreatedNotification(
+      // includes the file being updated
+      new S3notifications.LambdaDestination(usersRefresherLambdaFunction),
+      { prefix: `${STAGE}/${PERMISSIONS_FILENAME}` }
+    );
```

Comment:
Worth chatting with @jacobwinch here, who discovered CDK's S3 lambda triggers will deploy an intermediate, python based, lambda here (see also https://github.com/guardian/deploy-tools-platform/pull/579).

Would also recommend setting up [snapshot tests](https://github.com/guardian/cdk/blob/main/docs/best-practices.md#snapshot-testing) on this stack which will highlight this strangeness early on (i.e. before deploying).

Reply:
OMG you're right (see https://github.com/guardian/editorial-tools-pinboard/pull/118/commits/65888d8e2cda477727601491cbaa28bcc78dfe1e) I think I'll have to go down the SQS/SNS route as I really don't like this

Reply:
UPDATE: I paired with @jacobwinch and we deployed this branch as it stands... it turns out that the python lambda is only invoked at cloudformation time (presumably the APIs it needed to call to set up the notification on an existing bucket aren't available in CFN and so it needed this more complex runtime to execute those 🙈 ) so it's not quite as bad as we first thought (i.e. python lambda was some sort of proxy for each S3 event) although the lambda lives on beyond the cloudforming, so still messy!

FYI @akash1810 @jonathonherbert @davidfurey